src/JSON.ts

```diff
@@ -904,12 +905,9 @@ export class NullValue extends Value {
 export class ParseResult {
     private readonly _debugText: string; // Used only for debugging - copy of the original text being parsed
 
-    constructor(private _tokens: Token[], private _lineLengths: number[], private _value: Value | null, text: string, public readonly commentCount: number) {
-        assert(_tokens !== null);
-        assert(_tokens !== undefined);
-        assert(_lineLengths !== null);
-        assert(_lineLengths !== undefined);
-        assert(_value !== undefined);
+    constructor(private _tokens: Token[], private _lineLengths: number[], private _value: Value | undefined, text: string, public readonly commentCount: number) {
+        nonNullValue(_tokens, "_tokens");
```

Comment:
>_tokens [](start = 31, length = 7)

Nit: For the future, it would be nice to have some sort of nameof functionality, like [https://stackoverflow.com/questions/50470025/nameof-keyword-in-typescript](https://stackoverflow.com/questions/50470025/nameof-keyword-in-typescript)

Reply:
Yes, have been wishing for that.  The nonNullProp is the closest I was able to come - kind of solves the problem for object properties, but doesn't help with method parameters.  I'll have to look at https://github.com/dsherret/ts-nameof, but sounds a little expensive.